src/stomach.cpp

```diff
@@ -7,6 +7,60 @@
 #include "units.h"
 #include "game.h"
 #include "itype.h"
+#include "vitamin.h"
+
+void nutrients::min_in_place( const nutrients &r )
+{
+    kcal = std::min( kcal, r.kcal );
+    for( const std::pair<const vitamin_id, vitamin> &vit_pair : vitamin::all() ) {
+        const vitamin_id &vit = vit_pair.first;
+        int other = r.get_vitamin( vit );
+        if( other == 0 ) {
+            vitamins.erase( vit );
+        } else {
+            auto our_vit = vitamins.find( vit );
+            if( our_vit != vitamins.end() ) {
+                our_vit->second = std::min( our_vit->second, other );
+            }
+        }
+    }
+}
+
+void nutrients::max_in_place( const nutrients &r )
+{
+    kcal = std::max( kcal, r.kcal );
+    for( const std::pair<const vitamin_id, vitamin> &vit_pair : vitamin::all() ) {
+        const vitamin_id &vit = vit_pair.first;
+        int other = r.get_vitamin( vit );
+        if( other != 0 ) {
+            int &val = vitamins[vit];
+            val = std::max( val, other );
+        }
+    }
+}
+
+int nutrients::get_vitamin( const vitamin_id &vit ) const
+{
+    auto it = vitamins.find( vit );
+    if( it == vitamins.end() ) {
+        return 0;
+    }
+    return it->second;
+}
+
+bool nutrients::operator==( const nutrients &r ) const
+{
+    if( kcal != r.kcal ) {
```

Comment:
I do wonder why `return  kcal == r.kcal && vitamins == r.vitamins;` does not work?

Reply:
Because non-entries in the vitamins map have to be considered equivalent to zero entries.